### DISCUSSION TITLE
Excluding goals

### DISCUSSION THREAD

**Discussion by tf** [plausible/analytics#1451] [Anything Else]
I was wondering what it might take to extend the goal filtering functionality to also allow excluding certain goals. Our marketing website has a lot of visiters that only click the "log in" link to get to our backend. When looking at sign up conversion rates it would be nice to be able to exclude these visits since they cannot be expected to convert anyway - they are already customers.

In the code I see a couple of special cases to hide the "is not" filter type for goals. Is there a conceptual issue which would make this filter particularly hard to implement? Am I missing a more straightforward way to achieve what I am trying to do?

Thanks for working on Plausible! I really enjoy using it a lot so far! :clap:

**Comment by ukutaht** [comment]
Thanks for the request!

> In the code I see a couple of special cases to hide the "is not" filter type for goals. Is there a conceptual issue which would make this filter particularly hard to implement? Am I missing a more straightforward way to achieve what I am trying to do?

It's a bit trickier to accomplish on the backend (need to mess around with JOIN types) and since I didn't expect a big need for it, I left it unimplemented. Your use-case is definitely legitimate and I think the current limitation is pretty arbitrary from the user's perspective.

I can't think of a simpler way to accomplish this either.

Sorry I can't promise anything as we are prioritizing features that many people have asked for. But we can use this as the starting point and if anyone else has a need to exclude by goal, I'd be curious to hear about the use-case.

**Comment by metmarkosaric** [comment]
One more request for this today:

"I'm sure you get a billion feature requests all the time, but my # 1 would be to have negative goals. Subtract everyone who did this from the listing. The returning logins really pollute the data."

**Comment by ctessmer** [comment]
I have an online booking site—think hairstylists & their clients—for which I imagine I could use this exact feature.

Here's what I'd like to be able to do with Plausible:

1. Exclude all clients [of hairstylists] who ultimately visit `www.example.com/@usernme/*` links to book their appointment. Or segment their data in a different "bucket."
2. Exclude existing hairstylist customers [who are logging in to update their settings]

It sounds like this proposed feature by @tf solves this, so I'm a definite +1! 👍🏻 

*I never dared to try and figure this out in GA because of the overwhelm and intimidation of the platform. And because of that, my analytics have been useless for years. If Plausible fixes this—or if there is already a way to do this—it'd be a godsend for me!*

**Is there a way to do track these two user segments already in Plausible?**

FYI: I'm not yet a Plausible user, so this ^ may be a stupid question. But I'm planning on dropping the axe on GA very soon and switching and am elated that this is a feature even being considered. Thanks for reading! 😄

**Reply by RobertJoonas** [reply]
Thanks for sharing @ctessmer!

It sounds like you would indeed need to set up some goals and have the ability to exclude them. It's not possible to achieve that in any way at the current stage I'm afraid. We'd love to add this feature as it would make Plausible much more powerful, but unfortunately we cannot make any timeline promises at this point, due to our limited development resources. Sorry about that and thanks for understanding!

**Reply by ctessmer** [reply]
Thanks so much for the reply @RobertJoonas and for considering my idea. And of course I totally understand the inability to make any promises. In fact, you "let me down" so gracefully & kindly that I'm saving your response and adapting it for my own customers! 😄 So many features. So little time.

**Comment by Hricha-Shandily** [comment]
It is now possible to exclude goals in the Plausible dashboard, using the new "is not" filtering operator. Thanks for your ideas and support!

![exclude goal conversions in plausible](https://github.com/user-attachments/assets/661efc53-b2d2-4b5c-a95a-30e80355acfd)
